### DISCUSSION TITLE
Regular expression in python

### DISCUSSION THREAD

**Discussion by MahirFaisal07** [atom/atom#23273] [Q&A]
i am a beginner and only a few days into python,having difficulties in using regular expressions in python.

example:  

import re
z=open('mbox-short.txt')
for line in z:
    line=line.rstrip()
    if re.search('From:',line):
        print(line)

i wrote this code on atom and gave it a .py extension while saving. Then when i will try to run this code in command prompt , the code will traceback and give me  this massage below-

" C:\Users\Mahir\Desktop\py4e>re.py
Traceback (most recent call last):
  File "C:\Users\Mahir\Desktop\py4e\re.py", line 1, in <module>
    import re
  File "C:\Users\Mahir\Desktop\py4e\re.py", line 5, in <module>
    if re.search('From:',line):
AttributeError: partially initialized module 're' has no attribute 'search' (most likely due to a circular import)

C:\Users\Mahir\Desktop\py4e>  "

so looks like python isn't recognizing the 'import re' command

some is please help if i am doing something wrong or need to download some package and feature in atom to make this work.

i also attached photos for reference.

![re](https://user-images.githubusercontent.com/94616869/142416299-587f45d5-636c-44c6-9b73-ba3217416d25.PNG)
![re1](https://user-images.githubusercontent.com/94616869/142416404-712a7012-0ef6-43d7-84e5-e35dd94dd770.PNG)

**Comment by icecream17** [comment]
I think you're importing yourself, since you named the file re.py
Try renaming your file

**Reply by MahirFaisal07** [reply]
i tired, same thing happened.

 i think my atom isn't associated with 're' library, that's why this isn't working, and i don't know what to do to solve it

**Reply by icecream17** [reply]
idk either, but here's a stackoverflow: https://stackoverflow.com/questions/59762996/how-to-fix-attributeerror-partially-initialized-module
I can't access the stackoverflow bc of school, but look there and try things.
